src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -73,6 +73,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PrepareResourcesDependsOn>
   </PropertyGroup>
 
+  <!-- List of targets that emit DiagnosticMessages items -->
```

Comment:
Are all of the diagnostic messages intended to be around dependency resolution. It feels weird to have a list of targets that can warn be a requirement of ResolvePackageDependenciesDesignTime. Just about every target has the possibility of logging an error or warning that can go to the error list, but most should not run every time ResolvePackageDependenciesDesignTime.

If the pattern we want is for dependency-related diagnostics to use this and straight Log.LogError|Warning for the rest, then we should make the names for these Diagnostics less general: DependencyDiagnostic* or something.